tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -38,30 +51,38 @@ private static MockHttpMessageHandler CreateTokenResponseHttpHandlerWithX5CValid
                     // Check presence of client_assertion in request
                     Assert.IsTrue(formsData.TryGetValue("client_assertion", out string encodedJwt), "Missing client_assertion from request");
 
-                    // Check presence of x5c cert claim. It should exist.
+                    // Check presence and value of x5c cert claim.
                     var handler = new JwtSecurityTokenHandler();
                     var jsonToken = handler.ReadJwtToken(encodedJwt);
-                    Assert.IsTrue(jsonToken.Header.Any(header => header.Key == "x5c"), "x5c should be present");
+                    var x5c = jsonToken.Header.Where(header => header.Key == "x5c").FirstOrDefault();
+                    Assert.IsTrue((x5c.Key == "x5c"), "x5c should be present");
```

Comment:
should be: Assert.AreEqual("x5c", x5c.Key, "x5c should be present");
That way if the comparison fails, the unit test will tell you what it was versus what was expected (makes debugging easier)